includes/api/class-wc-rest-product-variations-controller.php

```diff
@@ -131,33 +131,32 @@ public function add_product_id( $args, $request ) {
 	 */
 	public function prepare_item_for_response( $post, $request ) {
 		$variation = wc_get_product( $post );
-		$post_data = get_post( $variation->get_variation_id() );
 		$data      = array(
-			'id'                 => $variation->get_variation_id(),
-			'date_created'       => wc_rest_prepare_date_response( $post_data->post_date_gmt ),
-			'date_modified'      => wc_rest_prepare_date_response( $post_data->post_modified_gmt ),
+			'id'                 => $variation->get_id(),
+			'date_created'       => wc_rest_prepare_date_response( $variation->get_date_created() ),
+			'date_modified'      => wc_rest_prepare_date_response( $variation->get_date_modified() ),
 			'description'        => $variation->get_variation_description(),
 			'permalink'          => $variation->get_permalink(),
 			'sku'                => $variation->get_sku(),
 			'price'              => $variation->get_price(),
 			'regular_price'      => $variation->get_regular_price(),
 			'sale_price'         => $variation->get_sale_price(),
-			'date_on_sale_from'  => $variation->sale_price_dates_from ? date( 'Y-m-d', $variation->sale_price_dates_from ) : '',
-			'date_on_sale_to'    => $variation->sale_price_dates_to ? date( 'Y-m-d', $variation->sale_price_dates_to ) : '',
+			'date_on_sale_from'  => $variation->get_date_on_sale_from() ? date( 'Y-m-d', $variation->get_date_on_sale_from() ) : '',
```

Comment:
Because `wc_rest_prepare_date_response` return data in the ISO8601/RFC3339 formatted datetime.

https://developer.wordpress.org/reference/functions/mysql_to_rfc3339/

> Parses and formats a MySQL datetime (Y-m-d H:i:s) for ISO8601/RFC3339.

Should we now return sale price dates with time too?